src/base/Range.hh

```diff
@@ -37,12 +37,138 @@ namespace celeritas
  * \endcode
  */
 
+//---------------------------------------------------------------------------//
+/*!
+ * Proxy container for iterating over a range of integral values.
+ *
+ * Here, T can be any of:
+ * - an integer,
+ * - an enum that has a "size_" member, or
+ * - an OpaqueId.
+ *
+ * It is OK to dereference the end iterator! The result should just be the
+ * off-the-end value for the range, e.g. `FooEnum::size_` or `bar.size()`.
```

Comment:
It won't crash, but wouldn't it be an arbitrary/random value?

Reply:
Not for the Range, since the end iterator is just an integer value, and not a pointer to anything.